### DISCUSSION TITLE
Converting a 2D problem into 3D

### DISCUSSION THREAD

**Discussion by sreejitsarmac** [idaholab/moose#26245] [Q&A Modules: Solid mechanics]
Hey guys. This is an example code from the MOOSE modules. This is a 2D subdomains problem. I am trying to turn this into a 3D problem. Can y'all tell me how to go about this? I'm getting errors on my codes. Thanks! Also can you tell me how to create a small circular subdomain on the 2D problem?

```
#
# Added subdomains and subdomain-specific properties
# https://mooseframework.inl.gov/modules/tensor_mechanics/tutorials/introduction/step03.html
#

[GlobalParams]
  displacements = 'disp_x disp_y'
[]

[Mesh]
  [generated]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 40
    ny = 20
    xmax = 2
    ymax = 1
  []

  # assign two subdomains
  [block1]
    type = SubdomainBoundingBoxGenerator
    input = generated
    block_id = 1
    bottom_left = '0 0 0'
    top_right = '1 1 0'
  []
  [block2]
    type = SubdomainBoundingBoxGenerator
    input = block1
    block_id = 2
    bottom_left = '1 0 0'
    top_right = '2 1 0'
  []
[]

[Modules/TensorMechanics/Master]
  [all]
    add_variables = true
  []
[]

[BCs]
  [bottom_x]
    type = DirichletBC
    variable = disp_x
    boundary = bottom
    value = 0
  []
  [bottom_y]
    type = DirichletBC
    variable = disp_y
    boundary = bottom
    value = 0
  []
  [Pressure]
    [top]
      boundary = top
      function = 1e7*t
    []
  []
[]

[Materials]
  [elasticity1]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 1e9
    poissons_ratio = 0.3
    block = 1
  []
  [elasticity2]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 5e8
    poissons_ratio = 0.3
    block = 2
  []
  [stress]
    type = ComputeLinearElasticStress
  []
[]

[Executioner]
  type = Transient
  solve_type = NEWTON
  petsc_options_iname = '-pc_type'
  petsc_options_value = 'lu'
  end_time = 5
  dt = 1
[]

[Outputs]
  exodus = true
[]
```

**Comment by GiudGiud** [comment]
Hello

I would stay:
- change the mesh to a 3D mesh by adding a nz and zmax parameter
- add a displacement disp_z
- everywhere you see disp_x add a similar object for disp_z

and that should make it 3D

Guillaume

**Reply by sreejitsarmac** [reply]
Hey Guillaume.

I did try those but it didn't work.  I think that the problem lies in the part:
```
[block1]
    type = SubdomainBoundingBoxGenerator
    input = generated
    block_id = 1
    bottom_left = '0 0 0'
    top_right = '1 1 0'
  []
  [block2]
    type = SubdomainBoundingBoxGenerator
    input = block1
    block_id = 2
    bottom_left = '1 0 0'
    top_right = '2 1 0'
```
Since I am unable to figure out how many more parameters I must add similar to bottom left and top right since in 3D we would need more of those. Please let me know what to do here.

Thanks
sreejitsarmac

**Reply by GiudGiud** [reply]
bottom_left and top_right are points. Right now they both a z coordinate of 0. This makes it 0 width in z. Maybe modify the Z coordinate of both these points to make it match the shape you want in 3D?

**Reply by sreejitsarmac** [reply]
Yeah I tried that as well. It still doesn't work. It gives me an error like "bad description"

**Reply by GiudGiud** [reply]
can you please paste the entire error?
and maybe also the current version of your input
